src/entropic/sinkhorn.jl

```diff
@@ -209,3 +222,113 @@ function sinkhorn2(μ, ν, C, ε, alg::Sinkhorn; regularization=false, plan=noth
+```
+where ``\\operatorname{W}_{ε}(μ,ν)``, ``\\operatorname{W}_{ε}(μ,μ)`` and
+``\\operatorname{W}_{ε}(ν,ν)`` are the entropically regularized optimal transport *cost*
+between `(μ,ν)`, `(μ,μ)` and `(ν,ν)`, respectively, defined as 
+```math
+\\operatorname{W}_{ε}(μ, ν) = \\langle C, γ^\\star \\rangle
+```
+where ``γ^\\star`` is the entropy-regularised transport plan between `μ` and `ν`. 
+The default algorithm for computing the term ``\\operatorname{OT}_{ε}(μ, ν)`` is the `SinkhornGibbs` algorithm.
+For the terms ``\\operatorname{OT}_{ε}(μ, μ)`` and ``\\operatorname{OT}_{ε}(ν, ν)``, the symmetric fixed point iteration of [^FeydyP19] is used. 
+Alternatively, a pre-computed optimal transport `plan` between `μ` and `ν` may be provided.  
+
+[^GPC18]: Aude Genevay, Gabriel Peyré, Marco Cuturi, Learning Generative Models with Sinkhorn Divergences, Proceedings of the Twenty-First International Conference on Artficial Intelligence and Statistics, (AISTATS) 21, 2018
+[^FeydyP19]: Jean Feydy, Thibault Séjourné, François-Xavier Vialard, Shun-ichi Amari, Alain Trouvé, and Gabriel Peyré. Interpolating between optimal transport and mmd using sinkhorn divergences. In The 22nd International Conference on Artificial Intelligence and Statistics, pages 2681–2690. PMLR, 2019.
+See also: [`sinkhorn2`](@ref)
+"""
+function sinkhorn_divergence(
+    μ::AbstractVecOrMat,
+    ν::AbstractVecOrMat,
+    C,
+    ε,
+    alg::Sinkhorn=SinkhornGibbs(),
+    algμ::Sinkhorn=SymmetricSinkhornGibbs(),
+    algν::Sinkhorn=SymmetricSinkhornGibbs();
+    regularization=true,
+    plan=nothing,
+    kwargs...,
+)
+    OTμν, OTμ, OTν = if (regularization == true) && (plan === nothing)
+        sinkhorn_loss(μ, ν, C, ε, alg; kwargs...),
+        sinkhorn_loss(μ, C, ε, algμ; kwargs...),
+        sinkhorn_loss(ν, C, ε, algν; kwargs...)
+    else
+        sinkhorn2(μ, ν, C, ε, alg; plan=plan, regularization=false, kwargs...),
+        sinkhorn2(μ, C, ε, algμ; plan=nothing, regularization=false, kwargs...),
+        sinkhorn2(ν, C, ε, algν; plan=nothing, regularization=false, kwargs...)
+    end
+    return max.(0, OTμν .- (OTμ .+ OTν) / 2)
+end
+"""
+    function sinkhorn_divergence(
+        μ,
+        ν,
+        Cμν,
+        Cμ,
+        Cν,
+        ε,
+        alg::Sinkhorn=SinkhornGibbs();
+        regularization=nothing,
+        plan=nothing,
+        kwargs...,
+    )
+Compute the Sinkhorn Divergence between finite discrete
+measures `μ` and `ν` with respect to the precomputed cost matrices `Cμν`,
+`Cμμ` and `Cνν`, entropic regularization parameter `ε` and algorithm `alg`.
+The default algorithm is the `SinkhornGibbs`.
+A pre-computed optimal transport `plan` between `μ` and `ν` may be provided.
+See also: [`sinkhorn2`](@ref) 
+"""
+function sinkhorn_divergence(
+    μ,
+    ν,
+    Cμν,
+    Cμ,
+    Cν,
+    ε,
+    alg::Sinkhorn=SinkhornGibbs(),
+    algμ::Sinkhorn=SymmetricSinkhornGibbs(),
+    algν::Sinkhorn=SymmetricSinkhornGibbs();
```

Comment:
I think it would be better to have a `SinkhornDivergence` algorithm that would wrap the individual algorithms, so that still only one algorithm is passed to the function.

Reply:
I'll implement something like `SinkhornDivergence(SinkhornGibbs(), SymmetricSinkhornGibbs(), SymmetricSinkhornGibbs())`

Reply:
Done!